### PR TITLE
Chapter2/homework/1

### PR DESCRIPTION
**Pull request by agakesik**
I think it took me way longer than it should xd

But I've refactored code from chapter1/lesson8 to be object oriented and made sure it passed all tests

### CODE REVIEW

File: Src/BootCamp.Chapter/Checks.cs [hunk 1]
```diff
@@ -9,31 +10,53 @@ public static class Checks
...
         public static void Clean(string file, string outputFile)
         {
+            var fileWithBalances = new FileWithBalances(file);
+            string cleanedContent = fileWithBalances.GetContent();
+            File.WriteAllText(outputFile, cleanedContent);
+        }
+
+        private static PeopleAndBalances CreatePeopleObjectFromArrayInput(string[] peopleAndBalances)
+        {
+            if (peopleAndBalances == null) peopleAndBalances = new string[0];
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: This line does 2 different thing, should be split into another line.
Please use brackets in order to explicitly say what happens if the if condition is true.

File: Src/BootCamp.Chapter/FileWithBalances.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            try
+            {
+                content = File.ReadAllText(pathToFile);
+            }
+            catch (Exception ex)
+            {
+                throw new ArgumentException("Path to source file does not exist", ex);
+            }
+
+            if (String.IsNullOrEmpty(content))
```
Comment (Almantask) [inline Src/BootCamp.Chapter/FileWithBalances.cs]: Please clarify the purpose of this. Looks redundant.

File: Src/BootCamp.Chapter/FileWithBalances.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        private static bool IsContentValid(string contentToValidate)
+        {
+            
+            var differentPeople = contentToValidate.Split(Environment.NewLine);
+
+            foreach (var person in differentPeople)
+            {
+                var nameAndBalances = person.Split(",");
+                if (!ISNameValid(nameAndBalances[0])) return false;
+                if (nameAndBalances.Length > 1 && !AreBalancesValid(nameAndBalances[1..^0])) return false;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/FileWithBalances.cs]: replace magic numbers with named constants.

File: Src/BootCamp.Chapter/FileWithBalances.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            return true;
+        }
+
+        private static bool ISNameValid(string name)
+        {
+            var nameRegex = new Regex(@"[a-zA-Z]+ [a-zA-Z]+'?-?[a-zA-Z]+[.]?$");
+
+            if (!nameRegex.IsMatch(name, 0))
+            {
+                Console.WriteLine(name);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/FileWithBalances.cs]: Side effect- should be removed.

Comment (Almantask) [reply Src/BootCamp.Chapter/FileWithBalances.cs]: Can be simplified:
```cs
private static bool ISNameValid(string name)
{
            var nameRegex = new Regex(@"[a-zA-Z]+ [a-zA-Z]+'?-?[a-zA-Z]+[.]?$");
            var isMatch = nameRegex.IsMatch(name, 0);
            return isMatch ;
}
```

File: Src/BootCamp.Chapter/FileWithBalances.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+            return true;
+        }
+
+        private static string Clean(string content)
+        {
+            var contentToClean = content;
+            contentToClean = contentToClean.Replace("_", "");
+
+            if (!IsContentValid(contentToClean)) throw new InvalidBalancesException("Balances file is still invalid after cleaning up the file");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/FileWithBalances.cs]: add brackets {}, do not inline throwing exceptions.

File: Src/BootCamp.Chapter/Person.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    highestBalance = balance;
+                }
+            }
+
+            return highestBalance;
+        }
+
+        public double GetBiggestLoss()
+        {
+            if (_balances.Count <= 1) return double.MinValue;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Person.cs]: why does it get compared 1

File: Src/BootCamp.Chapter/TextTable.cs [hunk 7]
```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class TextTable
+    {
+        /*
```
Comment (Almantask) [inline Src/BootCamp.Chapter/TextTable.cs]: Comment can be remvoed, was just an example

**Review by Almantask — CHANGES_REQUESTED** [review]
Looks good but could be better.

File: Src/BootCamp.Chapter/FileWithBalances.cs (hunk 2 above)
Comment (agakesik) [reply Src/BootCamp.Chapter/FileWithBalances.cs]: The idea was to not do validation and/or cleaning if input is null (it's the first time checking if content is null/empty), but actually I think I have a better way to structure it

**Review by Almantask — APPROVED** [review]
Good 😄

**Pull request closed without merging**